yarn-project/acir-simulator/src/public/index.test.ts

```diff
@@ -187,58 +187,74 @@ describe('ACIR public execution simulator', () => {
+        const parentEntryPointFn = ParentContractAbi.functions.find(f => f.name === 'pubEntryPoint')!;
+        const parentEntryPointFnSelector = keccak(Buffer.from(parentEntryPointFn.name)).subarray(0, 4);
+
+        const childContractAddress = AztecAddress.random();
+        const childValueFn = ChildContractAbi.functions.find(f => f.name === 'pubValue')!;
+        const childValueFnSelector = keccak(Buffer.from(childValueFn.name)).subarray(0, 4);
+
+        const initialValue = 3n;
+
+        const functionData = new FunctionData(parentEntryPointFnSelector, isInternal ?? false, false, false);
+        const args = encodeArguments(parentEntryPointFn, [
+          childContractAddress.toField().value,
+          toBigInt(childValueFnSelector),
+          initialValue,
+        ]);
 
-      const callContext = CallContext.from({
-        msgSender: AztecAddress.random(),
-        storageContractAddress: parentContractAddress,
-        portalContractAddress: EthAddress.random(),
-        isContractDeployment: false,
-        isDelegateCall: false,
-        isStaticCall: false,
-      });
+        const callContext = CallContext.from({
+          msgSender: AztecAddress.random(),
+          storageContractAddress: parentContractAddress,
+          portalContractAddress: EthAddress.random(),
+          isContractDeployment: false,
+          isDelegateCall: false,
+          isStaticCall: false,
+        });
+
+        // eslint-disable-next-line require-await
+        publicContracts.getBytecode.mockImplementation(async (addr: AztecAddress, selector: Buffer) => {
+          if (addr.equals(parentContractAddress) && selector.equals(parentEntryPointFnSelector)) {
+            return Buffer.from(parentEntryPointFn.bytecode, 'base64');
+          } else if (addr.equals(childContractAddress) && selector.equals(childValueFnSelector)) {
+            return Buffer.from(childValueFn.bytecode, 'base64');
+          } else {
+            return undefined;
+          }
+        });
+
+        publicContracts.getIsInternal.mockImplementation(() => {
+          return Promise.resolve(isInternal);
+        });
+
+        const execution: PublicExecution = { contractAddress: parentContractAddress, functionData, args, callContext };
+        const globalVariables = new GlobalVariables(new Fr(69), new Fr(420), new Fr(1), new Fr(7));
 
-      // eslint-disable-next-line require-await
-      publicContracts.getBytecode.mockImplementation(async (addr: AztecAddress, selector: Buffer) => {
-        if (addr.equals(parentContractAddress) && selector.equals(parentEntryPointFnSelector)) {
-          return Buffer.from(parentEntryPointFn.bytecode, 'base64');
-        } else if (addr.equals(childContractAddress) && selector.equals(childValueFnSelector)) {
-          return Buffer.from(childValueFn.bytecode, 'base64');
+        if (isInternal === undefined) {
+          // The error is don't seem to be propagated, but can see it in the logger.
```

Comment:
I think @benesjan waas making a list of errors and their propogations. Jan - another edge case!